### ISSUE TITLE
feat: ✨ support browser forced colors

### ISSUE THREAD

**Issue by smfonseca** [solid-design-system/solid#2347]
## User Story
As a user of Solid Design System, I want solid components to adapt to the color settings defined in my browser.

At the moment if I use Edge browser, and change "Page Colours" in the Accessibility tab, most components will become unusable due to lack of background color and other visual properties.

### Suggested Solution

Using a media query for [forced-colors](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors) and adjusting all components to render an outline or border when these modes are active may be the solution. 

Other relevant resources:

- https://www.w3.org/TR/css-color-3/#css-system 
- https://developer.mozilla.org/en-US/docs/Web/CSS/system-color

## DoR
- [ ] Item has business value
- [ ] Item all subtasks have been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified

## DoD
- [ ] All subtasks have been closed
